app/routes/action.js

```diff
@@ -116,5 +126,66 @@ module.exports = [
         })
       return h.response(filteredActions).code(OK_STATUS_CODE)
     }
+  },
+  {
+    method: 'POST',
+    path: '/action/{pathParam}/rule/',
+    options: {
+      validate: {
+        payload: Joi.object({
+          id: Joi.string().required(),
+          config: Joi.object().optional()
+        })
+      }
+    },
+    handler: (request, h) => commonHandler(request, h, (action, request, h) => {
+      const newRule = request.payload
+      addRule(action, newRule)
+      console.log(JSON.stringify(getActions(), null, 2))
+      return h.response({ message: 'Rule added successfully' }).code(OK_STATUS_CODE)
+    })
+  },
+  {
+    method: 'PUT',
+    path: '/action/{pathParam}/rule/',
+    options: {
+      validate: {
+        payload: Joi.object({
+          id: Joi.string().required(),
+          config: Joi.object().optional()
+        })
+      }
+    },
+    handler: (request, h) => commonHandler(request, h, (action, request, h) => {
+      const newRule = request.payload
+      const ruleIndex = findRuleIndex(action.eligibilityRules, newRule.id)
+      if (ruleIndex === -1) {
+        return h.response({ error: 'Rule not found' }).code(BAD_REQUEST_STATUS_CODE)
+      }
+      updateRule(action, ruleIndex, newRule)
+      console.log(JSON.stringify(getActions(), null, 2))
+      return h.response({ message: 'Rule updated successfully' }).code(OK_STATUS_CODE)
+    })
+  },
+  {
+    method: 'DELETE',
+    path: '/action/{pathParam}/rule/',
+    options: {
+      validate: {
+        payload: Joi.object({
+          id: Joi.string().required()
+        })
+      }
+    },
+    handler: (request, h) => commonHandler(request, h, (action, request, h) => {
+      const { id } = request.payload
+      const ruleIndex = findRuleIndex(action.eligibilityRules, id)
```

Comment:
The comment above on the `PUT` applies here too. Move the findRuleIndex invocation into land-action/index.js.
The handler should simply invoke `deleteRule(action, ruleId)`

Reply:
Done